tutoroars/templates/oars/jobs/init/superset-init-security.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+set -e
+
+#
+# Always install local overrides first
+#
+/usr/bin/env bash /app/docker/docker-bootstrap.sh
+
+# Set up a Row-Level Security filter to enforce course-based access restrictions.
+# Note: there are no cli commands or REST API endpoints to help us with this,
+# so we have to pipe python code directly into the superset shell. Yuck!
+superset shell <<EOF
+import logging
+from superset.connectors.sqla.models import (
+    RowLevelSecurityFilter, 
+    RLSFilterRoles,
+    SqlaTable,
+)
+from superset.utils.core import RowLevelSecurityFilterType
+from superset.extensions import security_manager
+from superset.migrations.shared.security_converge import Role
+
+session = security_manager.get_session()
+
+# Fetch the Open edX role
+role_name = "{{SUPERSET_OPENEDX_ROLE_NAME}}"
+openedx_role = session.query(Role).filter(Role.name == role_name).first()
+assert openedx_role, "{{SUPERSET_OPENEDX_ROLE_NAME}} role doesn't exist yet?"
+
+# Fetch the xapi table we want to restrict access to
+xapi_table = session.query(SqlaTable).filter(
+    SqlaTable.schema == "{{SUPERSET_CLICKHOUSE_XAPI_SCHEME}}"
+).filter(
+    SqlaTable.table_name == "{{SUPERSET_CLICKHOUSE_XAPI_TABLE}}"
+).first()
+assert xapi_table, "{{SUPERSET_CLICKHOUSE_XAPI_SCHEME}}.{{SUPERSET_CLICKHOUSE_XAPI_TABLE}} table doesn't exist yet?"
+
+# See if the Row Level Security Filter already exists
+group_key = "{{SUPERSET_XAPI_ROW_LEVEL_SECURITY_COURSE_ID_KEY}}"
+xapi_course_id_rls = (
+    session.query(
+        RowLevelSecurityFilter
+    ).filter(
+        RLSFilterRoles.c.role_id.in_((openedx_role.id,))
+    ).filter(
+        RowLevelSecurityFilter.group_key == group_key
+    )
+).first()
+
+# If it doesn't already exist, create one
+if xapi_course_id_rls:
+    create = False
+else:
+    create = True
+    xapi_course_id_rls = RowLevelSecurityFilter()
+
+# Sync the fields to our expectations
+xapi_course_id_rls.filter_type = RowLevelSecurityFilterType.REGULAR
+xapi_course_id_rls.group_key = "xapi_course_id"
+xapi_course_id_rls.tables = [xapi_table]
+# TODO FIXME -- how do we assign roles to this filter??
+#xapi_course_id_rls.roles = [openedx_role]
```

Comment:
RLS needs to be applied to the Open edX role -- have to figure out how to get SqlAlchemy to let me do that!!